system/jlib/jsocket.cpp

```diff
@@ -3179,6 +3207,40 @@ static bool decodeNumericIP(const char *text,unsigned *netaddr)
 
 static bool lookupHostAddress(const char *name,unsigned *netaddr)
 {
+    if (!name || !*name)
+        return false;
+
+    GetCachedHostName();
+
+    if (streq(name, "localhost"))
+    {
+        IpAddress ip;
+        ip.ipset(queryLocalIP());
+        ip.copyAddress(netaddr);
+        return true;
+    }
+    else if ( (streq(name, ".")) || (streq(name, myhostname.get())) )
+    {
+        IpAddress ip;
+        ip.ipset(queryHostIP());
+        ip.copyAddress(netaddr);
+        return true;
+    }
+    else if ( (!isdigit(name[0])) && (strchr(name, '.')) ) // fqdn name compare
+    {
+        char n1[256];
+        strcpy(n1, name);
+        int l1 = strcspn(n1, ".");
+        n1[l1] = '\0';
+        if (streq(n1, myhostname.get()))
+        {
+            IpAddress ip;
+            ip.ipset(queryHostIP());
+            ip.copyAddress(netaddr);
+            return true;
+        }
+    }
+
```

Comment:
You have to change lookupHostAddress() or else it will resolve your hostname to your loopback.

Reply:
Ok, I can see want it to resolve to same netaddr. as queryHostIP() would if dealing with the local hostname.

But it could do that by resolving (as it was), then checking if isLocal().
If it is, it could return queryHostIp()
That would allow any hostname that resolves to local (if there multiple) to continue to return the environment.conf specified interface= local IP.

Reply:
I will look into this.  But I did want to avoid calling getaddrinfo() when we can.